File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
while True:
    lst = list(map(int,input().split()))

    if lst[0]==0 and lst[1]==0 and lst[2]==0:
        break

    if max(lst) >= sum(lst) - max(lst):
        print("Invalid")
        continue

    if len(lst) == len(set(lst)):
        print("Scalene")
        continue

    if len(set(lst)) == 2:
        print("Isosceles")
        continue
    
    if len(set(lst)) == 1:
        print("Equilateral")
        continue

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.c
#include <stdio.h>

int main(void){
    while(1){
        int a,b,c;
        scanf("%d %d %d", &a, &b, &c);

        if(a==0 && b==0 && c==0){
            break;
        }
        if(a==b && b==c && c==a){
            printf("Equilateral\n");
            continue;
        }


        int max;
        if(a>b && a>c){
            max = a;
            if(a >= b+c){
                printf("Invalid\n");
                continue;
            }
        } else if (b>a && b>c){
            max = b;
            if(b >= a+c){
                printf("Invalid\n");
                continue;
            }
        } else {
            max = c;
            if(c >= a+b){
                printf("Invalid\n");
                continue;
            }
        }

        if (a==b || b==c || c==a){
            printf("Isosceles\n");
        } else {
            printf("Scalene\n");
        }

        

    }
    return 0;
}